engine/src/main/java/com/agiletec/aps/system/common/AbstractGenericCacheWrapper.java

```diff
@@ -119,7 +117,7 @@ protected <O> void manage(String key, O object, Action operation) {
             return;
         }
         Cache cache = this.getCache();
-        List<String> codes = (List<String>) this.get(cache, this.getCodesCacheKey(), List.class);
+        List<String> codes = new ArrayList<>(this.get(cache, this.getCodesCacheKey(), List.class));
```

Comment:
I would like a separate private method to make explicit/more readable what are you doing (`getCodesCopyFromCacheImmutable`) , just a suggesiont